optimade/server/exceptions.py

```diff
@@ -11,113 +12,71 @@
 )
 
 
-class StrReprMixin(HTTPException):
-    """This mixin can be useful when testing requires a string
-    representation of an exception that contains the HTTPException
+class HTTPException(FastAPIHTTPException, ABC):
+    """This abstract class makes it easier to subclass FastAPI's HTTPException with
+    new status codes.
+
+    It can also be useful when testing requires a string representation
+    of an exception that contains the HTTPException
     detail string, rather than the standard Python exception message.
+
     """
 
+    def __init__(self, detail: str = None, headers: dict = None) -> None:
+        super().__init__(status_code=self.status_code, detail=detail, headers=headers)
```

Comment:
You should set a standard `status_code`, e.g., `500`?

Reply:
Or maybe just define it and set it to `None`?

Reply:
Or is it already defined by the parent class?

Reply:
I was toying with being a bit more defensive here, so happy to add it

Reply:
Well the issue is that it's referring to a parameter it doesn't have?